packages/edit-post/src/components/header/header-toolbar/index.js

```diff
@@ -68,11 +75,48 @@ function HeaderToolbar() {
 		: /* translators: accessibility text for the editor toolbar when Top Toolbar is off */
 		  __( 'Document tools' );
 
+	const overflowItems = (
+		<>
+			{ isLargeViewport && (
+				<ToolbarItem
+					as={ ToolSelector }
+					showTooltip={ ! showIconLabels }
+					isTertiary={ showIconLabels }
+				/>
```

Comment:
More of a question for design, I wonder if it could be worth bringing the tools inline in the overflow menu instead of having them in an additional popover.

So instead of having something like:
<img width="334" alt="Screenshot 2020-08-13 at 4 11 28 pm" src="https://user-images.githubusercontent.com/677833/90110464-da396100-dd7f-11ea-939e-f6b86a141f3a.png">


It could be more like the Visual Editor / Code Editor options in the more menu in a separate section of the overflow menu:
<img width="264" alt="Screenshot 2020-08-13 at 4 11 45 pm" src="https://user-images.githubusercontent.com/677833/90110505-ec1b0400-dd7f-11ea-8136-9adfde7b339b.png">

Reply:
This relates to the earlier [conversation about hierarchy](https://github.com/WordPress/gutenberg/pull/24304#issuecomment-672964514). 
My only thought from a code standpoint is this will require quite a bit of markup/logic duplication, as we would have to use different components for each breakpoint.

Reply:
Separating these under section headings and using dividers could really help. I'll explore some ideas.